io/net/vlan_test.py

```diff
@@ -0,0 +1,351 @@
+        self.command = command
+        self.output = output
+        self.exitcode = exitcode
+
+    def __str__(self):
+        return "Command '%s' exited with %d.\nOutput:\n%s" \
+               % (self.command, self.exitcode, self.output)
+
+
+class VlanTest(Test):
+
+    """
+    :param switch_name: Switch name or IP
+    :param userid: userid of the switch to login into
+    :param password: password of the switch for user userid
+    :param vlan_num: vlan number where the port port_id will be added
+    :param host_port: host port id where the VLAN test will run
+    :param peer_port: peer port id where the VLAN test will run
+    :param host_intf: Host test N/W Interface
+    :param peer_intf: Peer test N/W Interface
+    :param peer_ip: IP address of peer
+    :param peer_user: Userid of the peer
+    :param peer_password: Password of the peer to ssh into
+    :param netmask: netmask of the test N/W Interfaces
+    """
+
+    def setUp(self):
+        """
+        test parameters
+        """
+        self.parameters()
+        self.switch_login(self.switch_name, self.userid, self.password)
+        self.peer_login(self.peer_ip, self.peer_user, self.peer_password)
+        self.get_ips()
+
+    def parameters(self):
+        self.switch_name = self.params.get("switch_name", '*', default=None)
+        self.userid = self.params.get("userid", '*', default=None)
+        self.password = self.params.get("password", '*', default=None)
+        self.vlan_num = self.params.get("vlan_num", '*', default=None)
+        self.host_port = self.params.get("host_port", '*', default=None)
+        self.peer_port = self.params.get("peer_port", '*', default=None)
+        self.host_intf = self.params.get("host_interface", '*', default=None)
+        self.peer_intf = self.params.get("peer_interface", '*', default=None)
+        self.peer_ip = self.params.get("peer_ip", '*', default=None)
+        self.peer_user = self.params.get("peer_user", '*', default=None)
+        self.peer_password = self.params.get("peer_password", '*',
+                                             default=None)
+        self.cidr_value = self.params.get("cidr_value", '*', default=None)
+        self.prompt = ">"
+
+    def switch_login(self, ip, username, password):
+        '''
+        telnet Login method for remote fc switch
+        '''
+        self.tnc = telnetlib.Telnet(ip)
+        self.tnc.read_until('username:')
+        self.tnc.write(username + '\n')
+        self.tnc.read_until('password:')
+        self.tnc.write(password + '\n')
+        ret = self.tnc.read_until(self.prompt)
+        assert self.prompt in ret
+
+    def _send_only_result(self, command, response):
+        output = response.splitlines()
+        if command in output[0]:
+            output.pop(0)
+        output.pop()
+        output = [element.lstrip()+'\n' for element in output]
+        response = ''.join(output)
+        response = response.strip()
+        self.log.info(''.join(response))
```

Comment:
self.log.info will log the message to both stdout and job log. I will check about avocado logging as it is i never tried till now. Thanks